bot.py

```diff
@@ -133,6 +166,144 @@ async def info(self, ctx: commands.Context[Any]) -> None:
         await ctx.reply(embed=constants.INFO_REPLY)
 
 
+P = ParamSpec("P")
+T = TypeVar("T")
+
+
+def only_from_guilds(*servers: list[int]) -> Callable[[Callable[P, T]], Callable[P, T]]:
+    """Add a check to make sure the command can only be run from one of a list of servers."""
+
+    # The lazy return type is due to discord.py not being more specific in their return type.
+    # `discord.Interaction` is currently a generic type, but is not documented that way.
+
+    def check_guild(itr: discord.Interaction) -> bool:  # type: ignore[type-arg]
+        return itr.guild_id in (servers or [])
+
+    return app_commands.check(check_guild)
+
+
+class ModCommands(commands.Cog):
+    __slots__ = ("bot",)
+
+    def __init__(self, sbot: MyBot) -> None:
+        self.bot = sbot
+
+    @app_commands.command()
+    @app_commands.default_permissions(manage_messages=True)
+    @only_from_guilds(*settings.discord.management_servers)
+    async def user_code(
+        self,
+        interaction: discord.Interaction,  # type: ignore[type-arg]
+        user: discord.User,
+        day: Annotated[AdventDay, app_commands.Range[int, 1, 25]],
+        part: Annotated[AdventPart, Literal[1, 2]],
```

Comment:
We _might_ want to make day or part optional. OTOH, if we expect >10 submissions per day per part, maybe we don't.